metricbeat/mb/mb.go

```diff
@@ -94,6 +104,44 @@ func (m *BaseModule) UnpackConfig(to interface{}) error {
 	return m.rawConfig.Unpack(to)
 }
 
+// ReConfigure re-configures the module with the given raw configuration
```

Comment:
Maybe add a comment about this method being intended to be used in module factories.

Reply:
Sure, will do. Again, just to further my own understanding of how Metricbeat modules work, what would be the downside / risk of calling this method from some place other than a module factory?

Reply:
Added in 168f7e53c.

Reply:
> Again, just to further my own understanding of how Metricbeat modules work, what would be the downside / risk of calling this method from some place other than a module factory?

Metricsets can access the module through `base.Module()`, one metricset could modify the base module, unexpectedly affecting other metricsets of the same module.